hacspec/src/array.rs

```diff
@@ -269,6 +269,242 @@ macro_rules! _array_base {
     };
 }
 
+
+#[macro_export]
+macro_rules! generic_array {
```

Comment:
Hm, this looks like a lot of code duplication... Not sure if we can do anything about that without making it more complex though.

Reply:
Indeed. The tradeoff here is that I can factor the code but that would make all the arrays generic If you want to define a monomorphic array you would have to define : 
```Rust
array!(StateGeneric,State, U32, 8)
```
You need `StateGeneric` because the macro system requires you to pass as arguments all the `ident` you will need in your type definition. Here we need the base `StateGeneric<T>` but also `type State = StateGeneric<U32>`. There is no way to hide `StateGeneric` to the end-user (same as `Canvas` for abstract integer).